packages/server/src/internalErrorClasses.ts

```diff
@@ -35,7 +35,8 @@ export class ValidationError extends GraphQLErrorWithCode {
   constructor(graphqlError: GraphQLError) {
     super(
       graphqlError.message,
-      ApolloServerErrorCode.GRAPHQL_VALIDATION_FAILED,
+      (graphqlError.extensions?.code as ApolloServerErrorCode) ??
```

Comment:
I think it would be simpler to just not use GraphQLErrorWithCode here and put this in the `extensions` below. (You would need to add the `this.name = this.constructor.name` or `this.name = 'ValidationError'` though; perhaps we should have a `NamedGraphQLError` superclass that just does this? This is really only for convenience when the errors are viewed in JS rather than serialized as GraphQL.)

Reply:
I made that change, it reads better to not extend GraphQLErrorWithCode.

Updated the test to validate the plugin-observed error matches what is serialized to the client.